File: output/GPT3.5-Turbo/xCodeEval_4000/Java/Go/9edead834db38c470d982b503582fb0f.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
	"strings"
)

var mod int = int(math.Pow(10, 9)) + 7
var n int
var m int
var a [][][]int
var st []string

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)

	n = nextInt(scanner)
	m = nextInt(scanner)
	a = make([][][]int, n)
	st = make([]string, n)

	for i := 0; i < n; i++ {
		st[i] = nextString(scanner)
	}

	ans := 0
	for i := 0; i < n; i++ {
		for j := 0; j < m; j++ {
			if st[i][j] == '1' {
				ans++
			}
		}
	}

	fmt.Println(ans)

}

func nextInt(scanner *bufio.Scanner) int {
	scanner.Scan()
	num, _ := strconv.Atoi(scanner.Text())
	return num
}

func nextString(scanner *bufio.Scanner) string {
	scanner.Scan()
	return scanner.Text()
}

File: output/GPT3.5-Turbo/xCodeEval_4000/Java/Python/9edead834db38c470d982b503582fb0f.py
import math

mod = int(math.pow(10, 9)) + 7
n = 0
m = 0
a = []
st = []

def main():
    global n, m, a, st
    n = int(input())
    m = int(input())
    a = [n//m] * m
    for i in range(1, n%m+1):
        a[i] += 1
    
    b = [0] * m
    for i in range(m):
        t = ((i)%m * (i)%m)%m
        b[t] += a[i]
    
    ans = 0
    for i in range(m):
        if i == 0:
            ans += (b[i]*(b[i]))
        else:
            ans += b[i] * b[m-i]
    
    print(ans)

if __name__ == "__main__":
    main()
